airflow/contrib/operators/sagemaker_create_tuning_job_operator.py

```diff
@@ -0,0 +1,120 @@
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from airflow.contrib.hooks.sagemaker_hook import SageMakerHook
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+from airflow.exceptions import AirflowException
+
+
+class SageMakerCreateHyperParameterTuningJobOperator(BaseOperator):
+
+    """
+       Initiate a SageMaker HyperParameter Tuning Job
+
+       This operator returns The ARN of the model created in Amazon SageMaker
+
+       :param region_name: The AWS region_name
+       :type region_name: string
+       :param tuning_job_config:
+       The configuration necessary to start a tuning job (templated)
+       :type tuning_job_config: dict
+       :param sagemaker_conn_id: The SageMaker connection ID to use.
+       :type sagemaker_conn_id: string
+       :param use_db_config: Whether or not to use db config
+       associated with sagemaker_conn_id.
+       If set to true, will automatically update the training config
+       with what's in db, so the db config doesn't need to
+       included everything, but what's there does replace the ones
+       in the training_job_config, so be careful
+       :type use_db_config: bool
+       :param aws_conn_id: The AWS connection ID to use.
+       :type aws_conn_id: string
+       :param wait_for_completion: if the operator should block
+       until training job finishes
+       :type wait_for_completion: bool
+       :param check_interval: if wait is set to be true, this is the time interval
+       which the operator will check the status of the training job
+       :type check_interval: int
+       :param max_ingestion_time: if wait is set to be true, the operator will fail
+       if the training job hasn't finish within the max_ingestion_time
+       (Caution: be careful to set this parameters because training can take very long)
+       :type max_ingestion_time: int
+
+       **Example**:
+           The following operator would start a training job when executed
+
+            sagemaker_tuning =
+               SageMakerCreateHyperParameterTuningJobOperator(
+                   task_id='sagemaker_tuning',
+                   training_job_config=config,
+                   job_name='my_sagemaker_tuning'
+                   sagemaker_conn_id='sagemaker_customers_conn'
+                   aws_conn_id='aws_customers_conn'
+               )
+       """
```

Comment:
In above comments, you need to change most of 'training's to 'tuning'.

Reply:
fixed